frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+		/// have funds to cover the deposit.
+		///
+		/// - `data`: A content of contract
+		///
+		/// Emits `ContractCreated`.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::create_contract(T::MaxContractContentLen::get()))]
+		pub fn create_contract(
+			origin: OriginFor<T>,
+			data: BoundedVec<u8, T::MaxContractContentLen>,
+			parties: BoundedVec<T::AccountId, T::MaxSignatures>,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin)?;
+
+			let required_deposit = Self::calculate_deposit(&data);
+			T::Currency::reserve_named(T::ReserveIdentifier::get(), &who, required_deposit)?;
+
+			let index = Self::contracts_count();
+			Contracts::<T, I>::insert(
+				index,
+				ContractDataStorage::<T::MaxContractContentLen, T::MaxSignatures, T::AccountId> {
+					data,
+					parties,
+					creator: who.clone(),
+				},
+			);
+
+			JudgeSignatures::<T, I>::insert(index, BoundedVec::default());
+			PartiesSignatures::<T, I>::insert(index, BoundedVec::default());
+
+			ContractsCount::<T, I>::put(index + 1);
+
+			Self::deposit_event(Event::ContractCreated { contract_id: index, creator: who });
+			Ok(())
+		}
+
+		/// Sign contract as party
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `contract_id`: The `id` of a contract that should be signed
+		///
+		/// Emits `PartieSigned`.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::party_sign_contract())]
+		pub fn party_sign_contract(
+			origin: OriginFor<T>,
+			contract_id: ContractIndex,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin)?;
+
+			let contracts =
+				Contracts::<T, I>::get(contract_id).ok_or(Error::<T, I>::ContractNotFound)?;
+			ensure!(contracts.parties.contains(&who), Error::<T, I>::NotParties);
+
+			PartiesSignatures::<T, I>::try_mutate(contract_id, |signatures_list: &mut Option<BoundedVec<T::AccountId, T::MaxSignatures>>| -> DispatchResult {
+				let signatures = signatures_list.as_mut().ok_or(Error::<T, I>::ContractNotFound)?;
+				let already_signed = signatures.contains(&who);
+
+				ensure!(!already_signed, Error::<T, I>::AlreadySigned);
+				signatures.try_push(who.clone()).map_err(|_| Error::<T, I>::TooMany)?;
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::PartieSigned { contract_id, signer: who });
+			Ok(())
+		}
+
+		/// Remove judge
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `judge`: Account that will be added as one of a judges
+		///
+		/// Emits `RemovedJudge`.
+		#[pallet::call_index(4)]
+		#[pallet::weight(100_000)]
+		pub fn remove_judge(origin: OriginFor<T>, judge: T::AccountId) -> DispatchResult {
+			T::AddJudgeOrigin::ensure_origin(origin)?;
+
+			Judges::<T, I>::mutate(|judges| judges.retain(|judge_account| judge_account != &judge));
```

Comment:
What about all signatures of this judge? Is it ok to leave them? (I guess so, but please confirm with stakeholders)

Reply:
Tricky question. Lets say yes, its ok to leave the signatures. This will require contract cleanup later, yes?